### ISSUE TITLE
npm-плагин для Phonegap Build

### ISSUE THREAD

**Issue by Ajaxy** [DrMoriarty/cordova-social-vk#13]
Привет.
Хочу попробовать использовать плагин при сборке в облаке через Phonegap Build.
Форкнул, поправил имя, версию и запушил в `npm`: https://github.com/Ajaxy/cordova-plugin-socialvk

При сборке в логах вылетают такие ошибки:

```
In file included from /project/Mozg/Plugins/cordova-plugin-socialvk/SocialVk.m:4:
Mozg/Plugins/cordova-plugin-socialvk/SocialVk.h:5:9: fatal error: 'VKSdk/VKSdk.h' file not found
#import <VKSdk/VKSdk.h>
        ^
1 error generated.
```

Видимо как-то неправильно устанавливается VKSdk.framework. Аналогичный плагин для фейсбука https://github.com/Wizcorp/phonegap-facebook-plugin/blob/master/plugin.xml работает нормально.

Есть ли идеи, в чём может быть проблема?
Спасибо.

**Comment by Ajaxy** [comment]
Возможно причина в том, что нужно вручную указать все зависимости для используемого фреймворка, как сделано тут:
https://github.com/Wizcorp/phonegap-facebook-plugin/blob/master/plugin.xml#L107
Не подскажете, какой список должен быть для вашего плагина?

**Comment by DrMoriarty** [comment]
Нет, дело не в зависимостях от стандартных фреймворков. В этом случае была бы совсем другая ошибка.
А вот ошибка с тем, что _иногда_ кордова не добавляет в файл проекта зависимость на VKSdk.framework действительно существует. И эта проблема проявляется безотносительно phonegap build, то есть при локальной сборке её тоже можно схватить.
Только вот локально с её последствиями легко справиться (просто добавить фреймворк в список зависимостей) а вот что делать на build сервере я даже не представляю.

**Comment by DrMoriarty** [comment]
Вот это решение не помогает случайно?
https://github.com/DrMoriarty/cordova-social-vk/issues/9

**Comment by Ajaxy** [comment]
Спасибо за быстрый ответ.

Кажется, проблема действительно связана с ссылками, их неправильно обрабатывает `npm`. В плагине для фейсбука была та же проблема: https://github.com/Wizcorp/phonegap-facebook-plugin#-----cordova-registry-warning-ios 

Можно ли как-то изменить структуру директорий, чтобы избежать наличия ссылок?

**Comment by Ajaxy** [comment]
Может быть вот этот комментарий как-то может помочь? https://github.com/Wizcorp/phonegap-facebook-plugin/issues/624#issuecomment-72519980

**Comment by DrMoriarty** [comment]
Возможно простое разыменование ссылок даст нужный эффект. (в mc при копировании есть такой флажок)
Если что, push request приветствуется.

**Comment by Ajaxy** [comment]
Да, простое перемещение ссылок помогло:

```
rm src/ios/VKSdk.framework/Headers src/ios/VKSdk.framework/VKSdk
mv src/ios/VKSdk.framework/Versions/Current/* src/ios/VKSdk.framework/
rm -rf src/ios/VKSdk.framework/Versions
```

**Comment by Ajaxy** [comment]
Правда возникла еще одна проблема. Я установил на свой телефон официальное приложение вконтакте, но плагин всё равно открывает WebView с формой логина и пароля от ВК. Не знаете, с чем может быть связано?

Авторизация через приложение по сути ключевой момент, иначе всю авторизацию и работу можно было бы реализовать через InAppBrowser и JS SDK.

**Comment by DrMoriarty** [comment]
А в приложении настроены схемы URL и всё такое? (они обычно подключаются плагином автоматически, но могут быть накладки)
Конкретно решение о логине через приложение или вебвью производится внутри src/ios/SocialVk.m
метод -(void)vkLoginWithPermissions:
Можно там брейкпоинт поставить и проверить, какой флаг выставляется для СДК.

**Comment by Ajaxy** [comment]
В приложении настроена URL-схема, но, кажется, это не должно влиять на выбор типа авторизации.

На самом деле неверно отрабатывает метод `[[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:VK_AUTHORIZE_URL_STRING]]`, причем как в вашем коде, так и в коде VKSdk: даже если [тут](https://github.com/DrMoriarty/cordova-social-vk/blob/master/src/ios/SocialVk.m#L159) задать переменную `inApp = NO`, VKSdk открывает Safari, а не VK App, так как [такая же проверка](https://github.com/VKCOM/vk-ios-sdk/blob/master/library/Source/VKSdk.m#L156) в их коде не выполняется.

**Comment by Ajaxy** [comment]
Ура, разобрался. Для iOS 9 надо было добавить

```
<key>LSApplicationQueriesSchemes</key>
<array>
    <string>vk</string>
    <string>vk-share</string>
    <string>vkauthorize</string>
</array>
```

https://github.com/VKCOM/vk-ios-sdk#using-sdk

**Comment by IYABAO** [comment]
@Ajaxy @DrMoriarty the same error
`fatal error: 'VKSdk/VKSdk.h' file not found
#import <VKSdk/VKSdk.h>
        ^
1 error generated.`

how to fix it anyway?
